extensions/wikibase/tests/src/org/openrefine/wikibase/qa/scrutinizers/FileNameScrutinizerTest.java

```diff
@@ -67,7 +67,67 @@ public void testLongFileName() throws IOException, MediaWikiApiErrorException {
         assertWarningsRaised(FileNameScrutinizer.fileNameTooLongType);
     }
 
-    @Test
+    @Test()
+    public void testValidCharactersInFilenameOdiaScript() throws IOException, MediaWikiApiErrorException {
+        MediaInfoEdit edit = new MediaInfoEditBuilder(TestingData.newMidA)
+                .addFileName("ଫାଇଲ.wav")
+                .build();
+
+        scrutinize(edit);
+        assertNoWarningRaised();
+    }
+
+    @Test()
+    public void testInvalidCharactersInFilenameTab() throws IOException, MediaWikiApiErrorException {
+        MediaInfoEdit edit = new MediaInfoEditBuilder(TestingData.newMidA)
+                .addFileName("Tabs (\t) are not allowed.png")
+                .build();
+
+        scrutinize(edit);
+        assertWarningsRaised(FileNameScrutinizer.invalidCharactersInFileNameType);
+    }
+
+    @Test()
+    public void testInvalidCharactersInFilenameNewLine() throws IOException, MediaWikiApiErrorException {
+        MediaInfoEdit edit = new MediaInfoEditBuilder(TestingData.newMidA)
+                .addFileName("New lines (\n) are not allowed.png")
+                .build();
+
+        scrutinize(edit);
+        assertWarningsRaised(FileNameScrutinizer.invalidCharactersInFileNameType);
+    }
+
+    @Test()
+    public void testInvalidCharactersInFilenameCarriageReturn() throws IOException, MediaWikiApiErrorException {
+        MediaInfoEdit edit = new MediaInfoEditBuilder(TestingData.newMidA)
+                .addFileName("Carriage returns (\r) are not allowed.png")
+                .build();
+
+        scrutinize(edit);
+        assertWarningsRaised(FileNameScrutinizer.invalidCharactersInFileNameType);
+    }
+
+    @Test()
+    public void testInvalidCharactersInFilenameFormFeed() throws IOException, MediaWikiApiErrorException {
+        MediaInfoEdit edit = new MediaInfoEditBuilder(TestingData.newMidA)
+                .addFileName("Form feeds (\f) are not allowed.png")
+                .build();
+
+        scrutinize(edit);
+        assertWarningsRaised(FileNameScrutinizer.invalidCharactersInFileNameType);
+    }
+
+    @Test()
+    public void testInvalidCharactersInFilenameBackspace() throws IOException, MediaWikiApiErrorException {
+        MediaInfoEdit edit = new MediaInfoEditBuilder(TestingData.newMidA)
+                .addFileName("Backspaces (\b) are not allowed.png")
+                .build();
+
+        scrutinize(edit);
+        assertWarningsRaised(FileNameScrutinizer.invalidCharactersInFileNameType);
+    }
+
+    @Test(enabled = false) // Vertical bars should be allowed
```

Comment:
Where did you see that vertical bars should be allowed? I just tried uploading a file to Commons with a vertical bar and it is refused.